content/scripts/document.js

```diff
@@ -7,23 +7,23 @@ function buildPath(contentPath, slug) {
   return path.join(contentPath, slugToFoldername(slug));
 }
 
-async function create(contentPath, html, meta, wikiHistory = null) {
-  const folder = buildPath(contentPath, meta.slug);
-  await fs.promises.mkdir(folder, { recursive: true });
+function create(contentPath, html, metadata, wikiHistory = null) {
+  const folder = buildPath(contentPath, metadata.slug);
+  fs.mkdirSync(folder, { recursive: true });
 
-  await fs.promises.writeFile(path.join(folder, "index.html"), html);
-
-  await fs.promises.writeFile(
-    path.join(folder, "index.yaml"),
-    yaml.safeDump(meta)
-  );
+  saveFile(path.join(folder, "index.html"), html, metadata);
 
   if (wikiHistory) {
-    await fs.promises.writeFile(
+    fs.writeFileSync(
       path.join(folder, "wikihistory.json"),
       JSON.stringify(wikiHistory, null, 2)
     );
   }
 }
 
-module.exports = { buildPath, create };
+function saveFile(filePath, html, metadata) {
+  const combined = `---\n${yaml.safeDump(metadata)}\n---\n${html.trim()}\n`;
```

Comment:
Nit. There's an extra line break after the `yaml.safeDump()`, for example:
```yaml
---
title: Game development
slug: Games
summary: >-
  Gaming is one of the most popular computer activities. New technologies are
  constantly arriving to make it possible to develop better and more powerful
  games that can be run in any standards-compliant web browser.
tags:
  - Apps
  - Game Development
  - Gamedev
  - Games
  - HTML5 Games
  - JavaScript Games
  - Web

---
```
```suggestion
  const combined = `---\n${yaml.safeDump(metadata)}---\n${html.trim()}\n`;
```